contracts/sfc/SFC.sol

```diff
@@ -901,6 +920,70 @@ contract SFC is Initializable, Ownable, Version {
         }
     }
 
+    /// Seal epoch - recalculate average uptime time of validators
+    function _sealEpochAverageUptime(
+        uint256 epochDuration,
+        EpochSnapshot storage snapshot,
+        EpochSnapshot storage prevSnapshot,
+        uint256[] memory validatorIDs,
+        uint256[] memory uptimes
+    ) internal {
+        for (uint256 i = 0; i < validatorIDs.length; i++) {
+            uint256 validatorID = validatorIDs[i];
+            // compute normalised uptime as a percentage in the Q1.30 fixed-point format
+            uint256 normalisedUptime = (uptimes[i] * (1 << 30)) / epochDuration;
+            if (normalisedUptime > 1 << 30) {
+                normalisedUptime = 1 << 30;
+            }
+            AverageUptime memory previous = prevSnapshot.averageUptime[validatorID];
+            AverageUptime memory current = _addElementIntoAverageUptime(uint32(normalisedUptime), previous);
+            snapshot.averageUptime[validatorID] = current;
+
+            // remove validator if average uptime drops below min average uptime
+            // (by setting minAverageUptime to zero, this check is ignored)
+            if (current.averageUptime < c.minAverageUptime() && current.epochs >= c.averageUptimeEpochsThreshold()) {
+                _setValidatorDeactivated(validatorID, OFFLINE_AVG_BIT);
+                _syncValidator(validatorID, false);
+            }
+        }
+    }
+
+    function _addElementIntoAverageUptime(
+        uint32 newValue,
+        AverageUptime memory prev
+    ) private view returns (AverageUptime memory) {
+        AverageUptime memory cur;
+        if (prev.epochs == 0) {
+            // the only element for the average
+            cur.averageUptime = uint32(newValue);
+            cur.epochs = 1;
+            return cur;
+        }
+
+        // the number of elements to calculate the average from
+        uint64 n = prev.epochs + 1;
+        // use lemma to add new value into the average
+        uint64 tmp = (n - 1) * uint64(prev.averageUptime) + uint64(newValue);
+
+        // apply remainder from the division in the previous iteration to reduce error
+        if (prev.remainder != 0) {
```

Comment:
Remove the condition and keep the statement inside the if-statement only.

Reply:
with using Decimal.unit(), the precision should be much higher, so the remainder should be meaningless anymore - removed.